test/tesla/adapter/gun_test.exs

```diff
@@ -186,14 +186,14 @@ defmodule Tesla.Adapter.GunTest do
     assert response.status == 500
   end
 
-  test "error on socks proxy" do
+  test "error waiting for response on socks proxy" do
```

Comment:
@yordis maybe we just can get rid of this test since socks proxy is already supported.

Reply:
I do not see why not, but what about `v1` people? I am just wondering if this will impose a major Tesla version bump.

Reply:
https://github.com/elixir-tesla/tesla/pull/587#issuecomment-1702687962

nevermind

Reply:
Oh you are right about the v1. In my opinion, I'd go for enforcing >= 2.0 and release a major version of tesla.

Reply:
😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭

Reply:
If we only change the error returned I don’t think we need a major version bump, let’s just make sure all tests pass for both versions of gun (we can match on {:error, _})

Reply:
But @teamon this is not the only change, the [`headers`](https://ninenines.eu/docs/en/gun/2.0/manual/gun.headers/) fn is not available in 1.0 so we will need to add an `if` use one function or other depending on the version. What about the tests here, how could you ensure that it runs against two versions?

Reply:
That's not a big problem:

```ex

if version_check do
  def dosmth(...), :gun.headers(...)
else
  def dosmth(...), :gun.other_fun(...)
end
```

Reply:
Welcome to the most beautiful feature of Elixir 😄 😄 😄 😄

Reply:
..... right 😭

Reply:
Oh! I see si you rely on a test mix.lock file, dark magic 🧙 . Cool, I'll do it in that way 😀

Reply:
Ok guys. Please take a look again. I thought about defining two modules based on the version and duplicating the whole files but, from now, I think that we can keep it like that as the differences are very small.

Notice that I'm doing the check more explicitly, using `Version` instead of `function_exported?`.

Reply:
PS: please, could you re-run the failed tests? I think they are flaky ones [test/tesla/middleware/retry_test.exs].